core/services/directrequest/delegate.go

```diff
@@ -98,22 +121,37 @@ func (l *listener) Start() error {
 	if !connected {
 		return errors.New("Failed to register listener with logBroadcaster")
 	}
-	l.unsubscribeLogs = unsubscribe
+	go l.run()
+
+	hbUnsubscribe := l.headBroadcaster.Subscribe(l)
+
+	l.shutdownWaitGroup.Add(1)
+	go func() {
+		defer hbUnsubscribe()
+		defer unsubscribeLogs()
+		defer l.shutdownWaitGroup.Done()
+		<-l.chStop
+	}()
+
 	return nil
 }
 
 // Close complies with job.Service
 func (l *listener) Close() error {
-	if l.unsubscribeLogs != nil {
-		l.unsubscribeLogs()
+	if !l.OkayToStop() {
```

Comment:
There's a better version of this. Check out `StartOnce` and `StopOnce` on the same type. They're safer against races, especially in the `Start` case. Probably should remove `OkayToStart/Stop`

Reply:
ah yes, keepers used OkayTo.. - ok changed and added a story: https://app.clubhouse.io/chainlinklabs/story/7237/replace-oktostart-and-oktostop-calls-with-startonce-stoponce